lib/net/ldap/connection.rb

```diff
@@ -652,4 +661,33 @@ def delete(args)
 
     pdu
   end
+
+  # Internal: Returns a Socket like object used internally to communicate with
+  # LDAP server.
+  #
+  # Typically a TCPSocket, but can be a OpenSSL::SSL::SSLSocket
+  def socket
+    return @conn if defined? @conn
```

Comment:
Out of curiosity, I have a question. 
Why is `defined?` used? it looks working as `if @conn`. but there are a little bit difference between the both. 

```ruby
@foo = 1 # <= defined instance variable
puts 'ok' if @foo # => 'ok'
puts 'ok' if defined? @foo # => 'ok'   so the both are same.

@bar = nil # defined but its value is nil
puts 'ok' if @bar # => nothing
puts 'ok' if defined @bar # => 'ok'  so different behavior
```
I believe that this difference doesn't affect on the behavior of `Net::LDAP`.